_posts/2023-04-19-12.0.0-release.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "Apache Arrow 12.0.0 Release"
+date: "2023-04-19 00:00:00"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+
+The Apache Arrow team is pleased to announce the 12.0.0 release. This covers
+over 3 months of development work and includes [**XXX resolved issues**][1]
+from [**YYY distinct contributors**][2]. See the [Install Page](https://arrow.apache.org/install/)
+to learn how to get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 11.0.0 release, Wang Mingming, Mustafa Akur and Ruihang Xia
+have been invited to be committers.
+Will Jones have joined the Project Management Committee (PMC).
+
+Thanks for your contributions and participation in the project!
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+
+## C++ notes
```

Comment:
@zeroshade @benibus @westonpace can you help with the release notes?

Reply:
```suggestion
## C++ notes
* The FixedShapeTensor Logical value type has been implemented using ExtensionType [GH-15483](https://github.com/apache/arrow/issues/15483)
* NaNs are correctly handled now for Parquet predicate push-downs [GH-18481](https://github.com/apache/arrow/issues/18481)
* Run-End Encoded Arrays have been implemented and are accessible [GH-32104](https://github.com/apache/arrow/issues/32104)
* Added support for DeltaLengthByteArray encoding to the Parquet writer [GH-33024](https://github.com/apache/arrow/issues/33024)
* Convenience Append(std::optional<T>...) methods have been added to array builders [GH-14863](https://github.com/apache/arrow/issues/14863)
* Added support for reading Parquet page indexes [GH-33596](https://github.com/apache/arrow/issues/33596) and writing page indexes [GH-34053](https://github.com/apache/arrow/issues/34053)
* Acero has been moved out of libarrow into it's own shared library, allowing for smaller builds of the core libarrow [GH-15280](https://github.com/apache/arrow/issues/15280)
* Parquet writer can write columns in parallel now [GH-33655](https://github.com/apache/arrow/issues/33655)
* GcsFileSystem::OpenInputFile avoids unnecessary downloads [GH-34051](https://github.com/apache/arrow/issues/34051)
* Fixed incorrect number of rows in Parquet V2 page headers [GH-34086](https://github.com/apache/arrow/issues/34086)
* Exec nodes now can have a concept of "ordering" and will reject non-sensible plans [GH-34136](https://github.com/apache/arrow/issues/34136)
* Added support for the union type in ORC writer [GH-34262](https://github.com/apache/arrow/issues/34262)
* Fixed incorrect Parquet page null_count when stats are disabled [GH-34326](https://github.com/apache/arrow/issues/34326)
* Improved handling of "logical" nulls such as with union and RunEndEncoded arrays [GH-34361](https://github.com/apache/arrow/issues/34361)
* A deprecated OpenTelemetry header was removed from the Flight library [GH-34417](https://github.com/apache/arrow/issues/34417)
* Fixed timestamp type mapping between ORC and arrow [GH-34590](https://github.com/apache/arrow/issues/34590)
* Fixed crash in "take" kernels on ExtensionArrays with an underlying dictionary type [GH-34619](https://github.com/apache/arrow/issues/34619)
* Added support for reading BloomFilters to the Parquet Reader [GH-34665](https://github.com/apache/arrow/issues/34665)
* Fixed bug where the C-Data bridge did not preserve nullability of map values on import [GH-34983](https://github.com/apache/arrow/issues/34983)
* Added support for EqualOptions to RecordBatch::Equals [GH-34968](https://github.com/apache/arrow/issues/34968)
* zstd dependency upgraded to v1.5.5 [GH-34899](https://github.com/apache/arrow/issues/34899)
* Parquet File-writer can now add additional key-value metadata after it has been opened [GH-34888](https://github.com/apache/arrow/issues/34888)
* Fixed ORC CHAR type mapping with Arrow [GH-34823](https://github.com/apache/arrow/issues/34823)
* Added FromTensor, ToTensor and strides methods to FixedShapeTensorArray [GH-34796](https://github.com/apache/arrow/issues/34796)
```

I think I got a good cross section of the release notes here without it being "too much". @benibus @felipecrv @westonpace feel free to add anything I forgot here. Thanks!

Reply:
+1. All significant changes I've merged are listed here.

Reply:
In addition to Matt's:
```suggestion
## C++ notes
* Added support for reading JSON datasets [GH-33209](https://github.com/apache/arrow/issues/33209)
* Fixed incorrect handling of uncompressed body buffers in IPC reader, added IpcWriteOptions::min_space_savings for optional compression optimizations [GH-15102](https://github.com/apache/arrow/issues/15102)
* Cast kernels are now built into libarrow by default [GH-34388](https://github.com/apache/arrow/issues/34388)
```